x-pack/test/alerting_api_integration/security_and_spaces/group1/tests/alerting/backfill/delete.ts

```diff
@@ -0,0 +1,328 @@
+            case 'superuser at space1':
+            // User has all privileges in this space
+            case 'space_1_all at space1':
+            // User has all privileges in this space
+            case 'space_1_all_with_restricted_fixture at space1':
+              expect(deleteResponse1.statusCode).to.eql(204);
+              expect(deleteResponse2.statusCode).to.eql(204);
+
+              await supertest
+                .get(
+                  `${getUrlPrefix(
+                    apiOptions.spaceId
+                  )}/internal/alerting/rules/backfill/${backfillId1}`
+                )
+                .set('kbn-xsrf', 'foo')
+                .expect(404);
+
+              await supertest
+                .get(
+                  `${getUrlPrefix(
+                    apiOptions.spaceId
+                  )}/internal/alerting/rules/backfill/${backfillId2}`
+                )
+                .set('kbn-xsrf', 'foo')
+                .expect(404);
+
+              try {
+                await getScheduledTask(backfillId1);
+                throw new Error('Should have removed scheduled task');
+              } catch (e) {
+                expect(e.meta.statusCode).to.eql(404);
+              }
+
+              try {
+                await getScheduledTask(backfillId2);
+                throw new Error('Should have removed scheduled task');
+              } catch (e) {
+                expect(e.meta.statusCode).to.eql(404);
+              }
+
+              break;
+            default:
+              throw new Error(`Scenario untested: ${JSON.stringify(scenario)}`);
+          }
+        });
+
+        it('should handle delete request appropriately when backfill does not exist', async () => {
+          // get backfill as current user
+          const response = await supertestWithoutAuth
+            .delete(
+              `${getUrlPrefix(apiOptions.spaceId)}/internal/alerting/rules/backfill/does-not-exist`
+            )
+            .set('kbn-xsrf', 'foo')
+            .auth(apiOptions.username, apiOptions.password);
+
+          // These should all be the same 404 response
+          switch (scenario.id) {
+            // User can't do anything in this space
+            case 'no_kibana_privileges at space1':
+            // User has no privileges in this space
+            case 'space_1_all at space2':
+            // User has read privileges in this space
+            case 'global_read at space1':
+            // User doesn't have access to actions but that doesn't matter for backfill jobs
+            case 'space_1_all_alerts_none_actions at space1':
+            // Superuser has access to everything
+            case 'superuser at space1':
+            // User has all privileges in this space
+            case 'space_1_all at space1':
+            // User has all privileges in this space
+            case 'space_1_all_with_restricted_fixture at space1':
+              expect(response.statusCode).to.eql(404);
```

Comment:
Shouldn't the status code for user has no privileges be 403, like for delete above?

Reply:
This is testing the response for when the backfill ID doesn't exist. In this case, we are unable to get any information about the backfill in order to check auth so we throw a 404.